CMakeLists.txt

```diff
@@ -4,6 +4,9 @@ project ("Xournal++" CXX C)
 
 ## Also update changelog in debian folder!
 set (PROJECT_VERSION "1.0.13")
+set (CPACK_PACKAGE_VERSION_MAJOR "1")
```

Comment:
I guess `PROJECT_VERSION` should be generated from the other three values.

Reply:
Before doing that we would need to change the release pipeline as it currently parses the project version variable for determining the correct tag.

Reply:
You can fetch the calculated value from `CMakeCache.txt` after CMake has run, if that's possible.

Reply:
The problem is that the release pipeline never sees a version of the repository where `cmake` ran. It only gets the build artifacts of our pipeline "release" and parses the `CMakeLists.txt` directly from the github repository.
But it shouldn't be a problem to parse the three `CPACK_PACKAGE_VERSION` variables and concat them.

Reply:
The other way round is also possible.

I just intended to highlight the redundancy, which could in the end cause inconsistencies.

Reply:
Yeah the redundancy promises issues later on. Whichever way we decide we only need to keep the release pipeline in mind.

Reply:
Where exactly does this version parsing happen?

Reply:
Check out the task `Set Variables` in https://dev.azure.com/xournalpp/xournalpp/_releaseDefinition?definitionId=1&_a=definition-tasks&environmentId=6

Reply:
One simple solution is to upload a file containing the version number (generated by cmake) as an artifact, read it in the nightly pipeline, and then delete it before making the release.

By the way, it should be possible to define the nightly pipeline with a YAML file (we can probably do that in another PR).

Reply:
That is actually just one step more. Parsing it from a generated file is as hard as parsing it from the `CMakeLists.txt` but we don't have to care about that file.

Reply:
The place where version information is defined doesn't matter, as you will always have to parse something. But thinking about this, I now consider defining the version in an extra file for my own applications as well. Conceptually it seems like a good idea.